src/controlplane/client/connector.go

```diff
@@ -0,0 +1,274 @@
+	t.TLSClientConfig = &tls.Config{InsecureSkipVerify: endpoint.InsecureSkipVerify}
+	client.Transport = t
+
+	if err := dp.configureAuthentication(client, endpoint); err != nil {
+		return nil, err
+	}
+
+	return client, nil
+}
+
+func (dp *defaultConnector) configureAuthentication(httpClient *http.Client, endpoint config.Endpoint) error {
+	if endpoint.Auth == nil {
+		return nil
+	}
+
+	switch {
+	case endpoint.Auth.Type == plainAuth:
+		dp.logger.Debugf("Plain auth is used for endpoint %q", endpoint.URL)
+
+	case strings.EqualFold(endpoint.Auth.Type, bearerAuth):
+		dp.logger.Debugf("kubernetes Bearer token auth is used for endpoint %q", endpoint.URL)
+		httpClient.Transport = transport.NewBearerAuthRoundTripper(dp.inClusterConfig.BearerToken, httpClient.Transport)
+
+	case strings.EqualFold(endpoint.Auth.Type, mTLSAuth):
+		tlsConfig, err := dp.getTLSConfigFromSecret(endpoint.Auth.MTLS)
+		if err != nil {
+			return fmt.Errorf("could not load TLS configuration: %w", err)
+		}
+
+		dp.logger.Debugf("MTLS auth is used for endpoint %q", endpoint.URL)
+		httpClient.Transport = &http.Transport{
+			TLSClientConfig: tlsConfig,
+		}
+
+	default:
+		return fmt.Errorf("authorization type not supported: %q", endpoint.Auth.Type)
+	}
+
+	return nil
+}
+
+func (dp *defaultConnector) getTLSConfigFromSecret(mTLSConfig *config.MTLS) (*tls.Config, error) {
+	if mTLSConfig == nil {
+		return nil, fmt.Errorf("mTLS config cannot be nil")
+	}
+
+	if mTLSConfig.TLSSecretName == "" {
+		return nil, fmt.Errorf("mTLS secret name cannot be empty")
+	}
+
+	namespace := mTLSConfig.TLSSecretNamespace
+	secretName := mTLSConfig.TLSSecretName
+	if namespace == "" {
+		dp.logger.Debugf("TLS Secret name configured, but not TLS Secret namespace. Defaulting to `default` namespace.")
+		namespace = "default"
+	}
+
+	secret, err := dp.kc.CoreV1().Secrets(namespace).Get(context.Background(), secretName, metav1.GetOptions{})
+	if err != nil {
+		return nil, fmt.Errorf("could not find secret %q containing TLS configuration: %w", secretName, err)
+	}
+
+	var cert, key, cacert []byte
+
+	var ok bool
+	if cert, ok = secret.Data["cert"]; !ok {
+		return nil, fmt.Errorf("could not find TLS certificate in `cert` field in secret %q", secretName)
+	}
+
+	if key, ok = secret.Data["key"]; !ok {
+		return nil, fmt.Errorf("could not find TLS key in `key` field in secret %q", secretName)
+	}
+
+	cacert, hasCACert := secret.Data["cacert"]
+	insecureSkipVerifyRaw, hasInsecureSkipVerify := secret.Data["insecureSkipVerify"]
+
+	if !hasCACert && !hasInsecureSkipVerify {
+		return nil, fmt.Errorf("both cacert and insecureSkipVerify are not set. One of them need to be set to be able to call ETCD metrics")
+	}
+
+	// insecureSkipVerify is set to false by default, and can be overridden with the insecureSkipVerify field
+	insecureSkipVerify := false
+	if hasInsecureSkipVerify {
+		insecureSkipVerify = strings.ToLower(string(insecureSkipVerifyRaw)) == "true"
```

Comment:
this is a bit obscure, why insecureSkipVerifyRaw is coming from the secret? Aren't the user setting it up in the config already?

Reply:
Talked offline

Reply:
This is certainly weird. I'd suggest to keep compatibility (overriding user config) but add a `log.Warn` and a `// TODO` saying the feature is deprecated and slated for removal soon.